core/include/moveit/task_constructor/container.h

```diff
@@ -53,6 +53,8 @@ class ContainerBase : public Stage
 
 	size_t numChildren() const;
 	Stage* findChild(const std::string& name) const;
+	Stage* front() const;
+	Stage* back() const;
```

Comment:
Instead of providing just front and back, I suggest exposing a variant of `ContainerBasePrivate::childByIndex()`.

Reply:
Thanks for the feedback! I will make that change

Reply:
change made in most recent commit